src/coreclr/jit/codegencommon.cpp

```diff
@@ -12528,3 +12534,83 @@ void CodeGenInterface::VariableLiveKeeper::dumpLvaVariableLiveRanges() const
 }
 #endif // DEBUG
 #endif // USING_VARIABLE_LIVE_RANGE
+
+void CodeGen::genSetRegsModifiedForPoisoning()
+{
+    // For ARM32 we need a callee-reserved register since the scratch registers may have args.
+#if defined(TARGET_ARM)
+    regSet.rsSetRegsModified(RBM_R4);
+#endif
+}
+
+//-----------------------------------------------------------------------------
+// genPoisonFrame: Generate code that places a recognizable value into address exposed variables.
+//
+// Remarks:
+//   This function emits code to poison address exposed non-zero-inited local variables. We expect this function
+//   to be called when emitting code for the scratch BB that comes right after the prolog.
+//   The variables are poisoned using 0xcccccccc. This value is stored in EAX/RAX on xarch and r/x0 on ARM,
+//   so we expect that this reg is not live.
+void CodeGen::genPoisonFrame(regMaskTP regLiveIn)
+{
+#if defined(TARGET_XARCH)
+    const regNumber immRegNum = REG_EAX;
```

Comment:
I would like to have them defined in the target*.h files, like `#define REG_FRAME_POISONING`.

Reply:
This has to be changed anyway since we can home arguments into registers in the prolog, depending on RA.

Reply:
Turns out there is already `REG_SCRATCH` defined that I will reuse for this.